client_linux.go

```diff
@@ -572,6 +572,48 @@ func (c *client) SetRings(rings Rings) error {
 	return err
 }
 
+// Channels fetches channels for a single interface.
+func (c *client) Channels(ifi Interface) (*Channels, error) {
+	cs, err := c.channels(0, ifi)
+	if err != nil {
+		return nil, err
+	}
+	if f := len(cs); f != 1 {
+		panicf("ethtool: unexpected number of Channels messages for request index: %d, name: %q: %d",
```

Comment:
Should we panic on invalid kernel responses? Perhaps better return error, even if it's a hardly possible scenario.

Reply:
controllers handle panics